tests/fixtures/hdfs.py

```diff
@@ -0,0 +1,195 @@
+        sdk_marathon.install_app(hdfsclient_app_def)
+
+        sdk_auth.kinit(HDFS_CLIENT_ID, keytab="hdfs.keytab", principal=GENERIC_HDFS_USER_PRINCIPAL)
+        hdfs_cmd("rm -r -skipTrash {}".format(HDFS_DATA_DIR))
+        hdfs_cmd("mkdir -p {}".format(HDFS_DATA_DIR))
+        hdfs_cmd("chown {}:users {}".format(ALICE_USER, HDFS_DATA_DIR))
+        yield
+
+    finally:
+        sdk_marathon.destroy_app(HDFS_CLIENT_ID)
+
+
+def hdfs_cmd(cmd):
+    rc, _, _ = sdk_cmd.marathon_task_exec(HDFS_CLIENT_ID, "bin/hdfs dfs -{}".format(cmd))
+    if rc != 0:
+        raise Exception("HDFS command failed with code {}: {}".format(rc, cmd))
+
+
+@pytest.fixture(scope='session')
+def configure_security_spark():
+    yield from utils.spark_security_session()
+
+
+@pytest.fixture(scope='session')
+def setup_history_server(hdfs_with_kerberos, setup_hdfs_client, configure_universe):
+    try:
+        sdk_auth.kinit(HDFS_CLIENT_ID, keytab="hdfs.keytab", principal=GENERIC_HDFS_USER_PRINCIPAL)
+        hdfs_cmd("rm -r -skipTrash {}".format(HDFS_HISTORY_DIR))
+        hdfs_cmd("mkdir {}".format(HDFS_HISTORY_DIR))
+        hdfs_cmd("chmod 1777 {}".format(HDFS_HISTORY_DIR))
+
+        sdk_install.uninstall(HISTORY_PACKAGE_NAME, HISTORY_SERVICE_NAME)
+        sdk_install.install(
+            HISTORY_PACKAGE_NAME,
+            HISTORY_SERVICE_NAME,
+            0,
+            additional_options={
+                "service": {
+                    "name": HISTORY_SERVICE_NAME,
+                    "user": SPARK_HISTORY_USER,
+                    "log-dir": "hdfs://hdfs{}".format(HDFS_HISTORY_DIR),
+                    "hdfs-config-url": "http://api.{}.marathon.l4lb.thisdcos.directory/v1/endpoints"
+                        .format(HDFS_SERVICE_NAME)
+                },
+                "security": {
+                    "kerberos": {
+                        "enabled": True,
+                        "krb5conf": HDFS_KRB5_CONF,
+                        "principal": GENERIC_HDFS_USER_PRINCIPAL,
+                        "keytab": KEYTAB_SECRET_PATH
+                    }
+                }
+            },
+            wait_for_deployment=False,  # no deploy plan
+            insert_strict_options=False)  # no standard service_account/etc options
+        yield
+
+    finally:
+        sdk_install.uninstall(HISTORY_PACKAGE_NAME, HISTORY_SERVICE_NAME)
+
+
+@pytest.fixture(scope='module')
+def kerberized_spark(setup_history_server, hdfs_with_kerberos, configure_security_spark, configure_universe):
+    try:
+        additional_options = {
+            "hdfs": {
+                "config-url": "http://api.{}.marathon.l4lb.thisdcos.directory/v1/endpoints".format(HDFS_SERVICE_NAME)
+            },
+            "security": {
+                "kerberos": {
+                    "enabled": True,
+                    "realm": sdk_auth.REALM,
+                    "kdc": {
+                        "hostname": hdfs_with_kerberos.get_host(),
```

Comment:
I think that we should explicitly use `kerberos_env` here.

Reply:
switched to `kerberos_env`